src/con_duct/cli.py

```diff
@@ -0,0 +1,474 @@
+duct is a lightweight wrapper that collects execution data for an arbitrary
+command. This command can be invoked as either 'duct' or 'con-duct run'.
+
+Execution data includes execution time, system information, and resource usage
+statistics of the command and all its child processes. It is intended to
+simplify the problem of recording the resources necessary to execute a command,
+particularly in an HPC environment.
+
+Resource usage is determined by polling (at a sample-interval).
+During execution, duct produces a JSON lines (see https://jsonlines.org) file
+with one data point recorded for each report (at a report-interval).
+
+limitations:
+  Duct uses session id to track the command process and its children, so it
+  cannot handle the situation where a process creates a new session.
+  If a command spawns child processes, duct will collect data on them, but
+  duct exits as soon as the primary process exits.
+
+environment variables:
+  Many duct options can be configured by environment variables (which are
+  overridden by command line options).
+
+  DUCT_LOG_LEVEL: see --log-level
+  DUCT_OUTPUT_PREFIX: see --output-prefix
+  DUCT_SUMMARY_FORMAT: see --summary-format
+  DUCT_SAMPLE_INTERVAL: see --sample-interval
+  DUCT_REPORT_INTERVAL: see --report-interval
+  DUCT_CAPTURE_OUTPUTS: see --capture-outputs
+  DUCT_MESSAGE: see --message
+"""
+
+
+class CustomHelpFormatter(argparse.ArgumentDefaultsHelpFormatter):
+    def _fill_text(self, text: str, width: int, _indent: str) -> str:
+        return "\n".join([textwrap.fill(line, width) for line in text.splitlines()])
+
+
+def create_common_parser() -> argparse.ArgumentParser:
+    """Create a parser with common arguments shared across all commands."""
+    parser = argparse.ArgumentParser(add_help=False)
+    parser.add_argument(
+        "-l",
+        "--log-level",
+        default=DEFAULT_LOG_LEVEL,
+        choices=("NONE", "CRITICAL", "ERROR", "WARNING", "INFO", "DEBUG"),
+        type=str.upper,
+        help="Level of log output to stderr, use NONE to entirely disable.",
+    )
+    parser.add_argument(
+        "-q",
+        "--quiet",
+        action="store_true",
+        help="[deprecated, use log level NONE] Disable duct logging output (to stderr)",
+    )
+    return parser
+
+
+def setup_logging(args: argparse.Namespace) -> None:
+    """Configure logging based on parsed arguments.
+
+    Handles both --log-level and --quiet flags, applying them consistently
+    across all subcommands.
+    """
+    log_level = args.log_level
+
+    # Handle deprecated --quiet flag
+    if args.quiet:
+        log_level = "NONE"
+
+    # Special case: NONE means disable all logging
+    if log_level == "NONE":
+        logging.disable(logging.CRITICAL)
+    else:
+        logging.basicConfig(
+            format="%(asctime)s [%(levelname)-8s] %(name)s: %(message)s",
+            datefmt="%Y-%m-%dT%H:%M:%S%z",
+            level=log_level,
+        )
+
+
+@dataclass
+class RunArguments:
```

Comment:
so we will have only `Run` arguments but nothing similar for any other command? 

If not, since we are not generally adhering to such design pattern, and now seems to be used only in the tests , where could simply be used via `args` directly. That would also minimize amount of all those heavy manual listing of all args in each different kind of invocation